### PR TITLE
Update Go Strings to list String Functions.

### PR DESCRIPTION
**Pull request by SSwiniarski**
### Description

I updated the Go Strings concept to introduce string functions, and added one string function term to start out the list.

### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/go/concepts/strings/terms/count/count.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+The **`Count()`** function returns the number of times a given substring appears in a given string without overlapping.
+
+## Syntax
+
+```pseudo
+strings.Count(str, sub)
+```
+
+Where `str` is the string being searched and `sub` is the substring being searched for. It returns an `int` as the number of non-overlapping occurrences of `sub` in `str`.
```
Comment (KTom101) [inline content/go/concepts/strings/terms/count/count.md:23]: Include what happens when `sub` is empty.

**Review by KTom101 — CHANGES_REQUESTED** [review]
@SSwiniarski I've requested one thing.

**Comment by SSwiniarski** [comment]
@KTom101, completed your suggestion.

**Review by KTom101 — APPROVED** [review]
@SSwiniarski Just one additional change; but this can be moved on to a second review.

File: content/go/concepts/strings/terms/count/count.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+The **`Count()`** function returns the number of times a given substring appears in a given string without overlapping.
+
+## Syntax
+
+```pseudo
+strings.Count(str, sub)
+```
+
+Where `str` is the string being searched and `sub` is the substring being searched for. It returns an `int` as the number of non-overlapping occurrences of `sub` in `str`. When `sub` is an empty string, `Count()` will count the boundaries before and after each character in `str`. (i.e. When `str` is "A" and `sub` is "", `Count()` returns 2. When `str` is "ABC" and `sub` is "", then `Count()` returns 4.)
```
Comment (CBID2) [inline content/go/concepts/strings/terms/count/count.md:23]: [Nitpick] Consider making the beginning part of the paragraph two separate sentences. That way, users can gain a better understanding of the roles of  `str`&  `sub` in the above code snippet. 
```suggestion
Where `str` is the string being searched.  `sub`, on the other hand, is the substring being searched for. It returns an `int` as the number of non-overlapping occurrences of `sub` in `str`. When `sub` is an empty string, `Count()` will count the boundaries before and after each character in `str`. (i.e. When `str` is "A" and `sub` is "", `Count()` returns 2. When `str` is "ABC" and `sub` is "", then `Count()` returns 4.)
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @SSwiniarski! :) You've always given good feedback on my pull requests, so I'd thought I'd return the favor by doing a review for you! 😉 Anywho, your entry is off to a great start. I just gave a suggestion on how to improve one of your explanations. I hope it helps! 😊🙏🏽

**Review by caupolicandiaz — APPROVED** [review]
@SSwiniarski Overall, looks good. My one minor suggestion would be to change the example in the strings.md entry to only show the import syntax, or change the example to avoid having a duplicate.

**Comment by yangc95** [comment]
@SSwiniarski live links: Go [Strings](https://www.codecademy.com/resources/docs/go/strings): [Count()](https://www.codecademy.com/resources/docs/go/strings/count)

**Pull request merged**